### ISSUE TITLE
Tutorial: Cabal new-build fails

### ISSUE THREAD

**Issue by ole-magnus** [haskell-servant/servant#1176]
I tried out running the tutorial. Fails at first try.

`git clone https://github.com/haskell-servant/servant.git` ✅ 
`cd servant` ✅ 
`cabal new-build tutorial` ❌ 

output from `cabal new-build-tutorial`:

```
In order, the following will be built (use -v for more details):
 - aeson-1.4.3.0 (lib) --enable-library-profiling (requires build)
 - charset-0.3.7.1 (lib:charset) --enable-library-profiling (requires build)
 - http-api-data-0.4 (lib:http-api-data) --enable-library-profiling (requires build)
 - semigroupoids-5.3.2 (lib:semigroupoids) --enable-library-profiling (requires build)
 - vault-0.3.1.2 (lib:vault) --enable-library-profiling (requires build)
 - aeson-pretty-0.8.7 (lib) --enable-library-profiling (requires build)
 - free-5.1 (lib) --enable-library-profiling (requires build)
 - wai-3.2.2 (lib) --enable-library-profiling (requires build)
 - servant-0.16 (lib:servant) --enable-library-profiling (first run)
 - adjunctions-4.4 (lib) --enable-library-profiling (requires build)
 - warp-3.2.27 (lib) --enable-library-profiling (requires build)
 - wai-logger-2.3.4 (lib:wai-logger) --enable-library-profiling (requires build)
 - servant-client-core-0.16 (lib) --enable-library-profiling (first run)
 - kan-extensions-5.2 (lib:kan-extensions) --enable-library-profiling (requires build)
 - wai-extra-3.0.25 (lib) --enable-library-profiling (requires build)
 - servant-client-0.16 (lib) --enable-library-profiling (first run)
 - lens-4.17.1 (lib:lens) --enable-library-profiling (requires build)
 - wai-app-static-3.1.6.3 (lib) --enable-library-profiling (requires build)
 - hspec-wai-0.9.2 (lib) --enable-library-profiling (requires build)
 - servant-foreign-0.15 (lib) --enable-library-profiling (first run)
 - servant-docs-0.11.3 (lib) --enable-library-profiling (first run)
 - servant-server-0.16 (lib:servant-server) --enable-library-profiling (first run)
 - servant-js-0.9.4 (lib) --enable-library-profiling (requires build)
 - tutorial-0.10 (lib) --enable-library-profiling (first run)
 - tutorial-0.10 (test:spec) --enable-library-profiling (first run)
Configuring charset-0.3.7.1 (all, legacy fallback)...
Configuring http-api-data-0.4 (all, legacy fallback)...
Configuring semigroupoids-5.3.2 (all, legacy fallback)...
Configuring aeson-1.4.3.0 (lib)...
Building charset-0.3.7.1 (all, legacy fallback)...
Building aeson-1.4.3.0 (lib)...
Building http-api-data-0.4 (all, legacy fallback)...
Building semigroupoids-5.3.2 (all, legacy fallback)...
Configuring vault-0.3.1.2 (all, legacy fallback)...
Building vault-0.3.1.2 (all, legacy fallback)...

Failed to build aeson-1.4.3.0.
Build log (
/Users/myuser/.cabal/logs/ghc-8.2.1/aeson-1.4.3.0-bad4328997929bbe9e50d30ff755a57299c77d9b040fe160690c21d088b518b6.log
):
Configuring library for aeson-1.4.3.0..
Preprocessing library for aeson-1.4.3.0..
Building library for aeson-1.4.3.0..
[ 1 of 25] Compiling Data.Aeson.Compat ( Data/Aeson/Compat.hs, dist/build/Data/Aeson/Compat.o )
[ 2 of 25] Compiling Data.Aeson.Internal.Functions ( Data/Aeson/Internal/Functions.hs, dist/build/Data/Aeson/Internal/Functions.o )
[ 3 of 25] Compiling Data.Aeson.Parser.UnescapePure ( pure/Data/Aeson/Parser/UnescapePure.hs, dist/build/Data/Aeson/Parser/UnescapePure.o )
[ 4 of 25] Compiling Data.Aeson.Parser.Unescape ( Data/Aeson/Parser/Unescape.hs, dist/build/Data/Aeson/Parser/Unescape.o )
[ 5 of 25] Compiling Data.Aeson.Types.Generic ( Data/Aeson/Types/Generic.hs, dist/build/Data/Aeson/Types/Generic.o )
[ 6 of 25] Compiling Data.Aeson.Types.Internal ( Data/Aeson/Types/Internal.hs, dist/build/Data/Aeson/Types/Internal.o )
[ 7 of 25] Compiling Data.Aeson.Parser.Internal ( Data/Aeson/Parser/Internal.hs, dist/build/Data/Aeson/Parser/Internal.o )
[ 8 of 25] Compiling Data.Aeson.Parser ( Data/Aeson/Parser.hs, dist/build/Data/Aeson/Parser.o )
[ 9 of 25] Compiling Data.Attoparsec.Time.Internal ( attoparsec-iso8601/Data/Attoparsec/Time/Internal.hs, dist/build/Data/Attoparsec/Time/Internal.o )

attoparsec-iso8601/Data/Attoparsec/Time/Internal.hs:24:1: warning: [-Wunused-imports]
    The import of ‘Unsafe.Coerce’ is redundant
      except perhaps to import instances from ‘Unsafe.Coerce’
    To import instances alone, use: import Unsafe.Coerce()
   |
24 | import Unsafe.Coerce (unsafeCoerce)
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[10 of 25] Compiling Data.Attoparsec.Time ( attoparsec-iso8601/Data/Attoparsec/Time.hs, dist/build/Data/Attoparsec/Time.o )
[11 of 25] Compiling Data.Aeson.Parser.Time ( Data/Aeson/Parser/Time.hs, dist/build/Data/Aeson/Parser/Time.o )
[12 of 25] Compiling Data.Aeson.Types.FromJSON ( Data/Aeson/Types/FromJSON.hs, dist/build/Data/Aeson/Types/FromJSON.o )

Data/Aeson/Types/FromJSON.hs:490:3: warning: [-Winline-rule-shadowing]
    Rule "FromJSONKeyCoerce: fmap id" may never fire
      because ‘id’ might inline first
    Probable fix: add an INLINE[n] or NOINLINE[n] pragma for ‘id’
    |
490 |   "FromJSONKeyCoerce: fmap id"     forall (x :: FromJSONKeyFunction a).
    |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...

Data/Aeson/Types/FromJSON.hs:756:35: warning: [-Wunused-matches]
    Defined but not used: ‘v’
    |
756 | withBoundedScientific_ whenFail f v@(Number scientific) =
    |                                   ^

Data/Aeson/Types/FromJSON.hs:761:5: warning: [-Wname-shadowing]
    This binding for ‘exponent’ shadows the existing binding
      imported from ‘Prelude.Compat’ at Data/Aeson/Types/FromJSON.hs:82:1-21
      (and originally defined in ‘GHC.Float’)
    |
761 |     exponent = base10Exponent scientific
    |     ^^^^^^^^
ghc: panic! (the 'impossible' happened)
  (GHC version 8.2.1 for x86_64-apple-darwin):
        getUnboxedSumName
  513
  Call stack:
      CallStack (from HasCallStack):
        prettyCurrentCallStack, called at compiler/utils/Outputable.hs:1133:58 in ghc:Outputable
        callStackDoc, called at compiler/utils/Outputable.hs:1137:37 in ghc:Outputable
        pprPanic, called at compiler/prelude/KnownUniques.hs:104:5 in ghc:KnownUniques

Please report this as a GHC bug:  http://www.haskell.org/ghc/reportabug


Failed to build charset-0.3.7.1.
Build log (
/Users/myuser/.cabal/logs/ghc-8.2.1/charset-0.3.7.1-e29315645d173fb9b9cba89f3102ba42460e9d9c76f034f45cb8697007fa6688.log
):
Configuring charset-0.3.7.1...
Preprocessing library for charset-0.3.7.1..
Building library for charset-0.3.7.1..
[1 of 9] Compiling Data.CharSet.ByteSet ( src/Data/CharSet/ByteSet.hs, dist/build/Data/CharSet/ByteSet.o )
[2 of 9] Compiling Data.CharSet     ( src/Data/CharSet.hs, dist/build/Data/CharSet.o )
[3 of 9] Compiling Data.CharSet.Common ( src/Data/CharSet/Common.hs, dist/build/Data/CharSet/Common.o )
[4 of 9] Compiling Data.CharSet.Posix.Ascii ( src/Data/CharSet/Posix/Ascii.hs, dist/build/Data/CharSet/Posix/Ascii.o )
ghc: panic! (the 'impossible' happened)
  (GHC version 8.2.1 for x86_64-apple-darwin):
        getUnboxedSumName
  513
  Call stack:
      CallStack (from HasCallStack):
        prettyCurrentCallStack, called at compiler/utils/Outputable.hs:1133:58 in ghc:Outputable
        callStackDoc, called at compiler/utils/Outputable.hs:1137:37 in ghc:Outputable
        pprPanic, called at compiler/prelude/KnownUniques.hs:104:5 in ghc:KnownUniques

Please report this as a GHC bug:  http://www.haskell.org/ghc/reportabug


Failed to build http-api-data-0.4.
Build log (
/Users/myuser/.cabal/logs/ghc-8.2.1/http-api-data-0.4-54767e2e6255833572d69c7b05141b4918bd6a8cb8462a19f8107a4e9a3686ca.log
):
[1 of 1] Compiling Main             ( /Users/myuser/Dev/haskell/servant/dist-newstyle/tmp/src-38368/http-api-data-0.4/dist/setup/setup.hs, /Users/myuser/Dev/haskell/servant/dist-newstyle/tmp/src-38368/http-api-data-0.4/dist/setup/Main.o )
Linking /Users/myuser/Dev/haskell/servant/dist-newstyle/tmp/src-38368/http-api-data-0.4/dist/setup/setup ...
Configuring http-api-data-0.4...
Preprocessing library for http-api-data-0.4..
Building library for http-api-data-0.4..
[1 of 4] Compiling Web.Internal.HttpApiData ( src/Web/Internal/HttpApiData.hs, dist/build/Web/Internal/HttpApiData.o )
[2 of 4] Compiling Web.Internal.FormUrlEncoded ( src/Web/Internal/FormUrlEncoded.hs, dist/build/Web/Internal/FormUrlEncoded.o )
Encoded.o )
ghc: panic! (the 'impossible' happened)
  (GHC version 8.2.1 for x86_64-apple-darwin):
        getUnboxedSumName
  513
  Call stack:
      CallStack (from HasCallStack):
        prettyCurrentCallStack, called at compiler/utils/Outputable.hs:1133:58 in ghc:Outputable
        callStackDoc, called at compiler/utils/Outputable.hs:1137:37 in ghc:Outputable
        pprPanic, called at compiler/prelude/KnownUniques.hs:104:5 in ghc:KnownUniques

Please report this as a GHC bug:  http://www.haskell.org/ghc/reportabug


Failed to build semigroupoids-5.3.2.
Build log (
/Users/myuser/.cabal/logs/ghc-8.2.1/semigroupoids-5.3.2-6845e827b4c4de09c6796654f2debaf417d8bc0e7045d57bb58dd93877c13278.log
):
[1 of 1] Compiling Main             ( /Users/myuser/Dev/haskell/servant/dist-newstyle/tmp/src-38369/semigroupoids-5.3.2/dist/setup/setup.hs, /Users/myuser/Dev/haskell/servant/dist-newstyle/tmp/src-38369/semigroupoids-5.3.2/dist/setup/Main.o )
Linking /Users/myuser/Dev/haskell/servant/dist-newstyle/tmp/src-38369/semigroupoids-5.3.2/dist/setup/setup ...
Configuring semigroupoids-5.3.2...
Preprocessing library for semigroupoids-5.3.2..
Building library for semigroupoids-5.3.2..
[ 1 of 21] Compiling Data.Functor.Extend ( src/Data/Functor/Extend.hs, dist/build/Data/Functor/Extend.o )
[ 2 of 21] Compiling Data.Functor.Bind.Class ( src/Data/Functor/Bind/Class.hs, dist/build/Data/Functor/Bind/Class.o )
ghc: panic! (the 'impossible' happened)
  (GHC version 8.2.1 for x86_64-apple-darwin):
        getUnboxedSumName
  513
  Call stack:
      CallStack (from HasCallStack):
        prettyCurrentCallStack, called at compiler/utils/Outputable.hs:1133:58 in ghc:Outputable
        callStackDoc, called at compiler/utils/Outputable.hs:1137:37 in ghc:Outputable
        pprPanic, called at compiler/prelude/KnownUniques.hs:104:5 in ghc:KnownUniques

Please report this as a GHC bug:  http://www.haskell.org/ghc/reportabug


Failed to build vault-0.3.1.2.
Build log (
/Users/myuser/.cabal/logs/ghc-8.2.1/vault-0.3.1.2-0ff4956986264907561b289f7eec29b3a923f9cc973ec88c4646bcb516058535.log
):
Configuring vault-0.3.1.2...
Preprocessing library for vault-0.3.1.2..
Building library for vault-0.3.1.2..
[1 of 5] Compiling Data.Unique.Really ( src/Data/Unique/Really.hs, dist/build/Data/Unique/Really.o )
[2 of 5] Compiling Data.Vault.ST.Lazy ( src/Data/Vault/ST/Lazy.hs, dist/build/Data/Vault/ST/Lazy.o )
ghc: panic! (the 'impossible' happened)
  (GHC version 8.2.1 for x86_64-apple-darwin):
        getUnboxedSumName
  513
  Call stack:
      CallStack (from HasCallStack):
        prettyCurrentCallStack, called at compiler/utils/Outputable.hs:1133:58 in ghc:Outputable
        callStackDoc, called at compiler/utils/Outputable.hs:1137:37 in ghc:Outputable
        pprPanic, called at compiler/prelude/KnownUniques.hs:104:5 in ghc:KnownUniques

Please report this as a GHC bug:  http://www.haskell.org/ghc/reportabug
```

INFO:
```
▶ cabal --version
cabal-install version 2.0.0.0
compiled using version 2.0.0.2 of the Cabal library
```

```
▶ sw_vers
ProductName:    Mac OS X
ProductVersion: 10.14
BuildVersion:   18A391
```

```
▶ ghc --version
The Glorious Glasgow Haskell Compilation System, version 8.2.1
```

**Comment by arianvp** [comment]
You seem to hit the following bug in GHC https://gitlab.haskell.org/ghc/ghc/issues/14051

Which has been fixed in in GHC 8.2.2 

Could you try updating to GHC 8.2.2 (or newer) and try again?

**Comment by ole-magnus** [comment]
@arianvp 
Thanks for the quick response. Upgrading GHC indeed helps! Thank you.
